test/unit/app/models/statistics_announcement_test.rb

```diff
@@ -292,6 +270,107 @@ class StatisticsAnnouncementTest < ActiveSupport::TestCase
     assert_equal statistics_announcement_date3.id, statistics_announcement.reload.current_release_date_id
   end
 
+  test "only valid when associated publication is of a matching type" do
+    statistics          = create(:draft_statistics)
+    national_statistics = create(:draft_national_statistics)
+    policy_paper        = create(:draft_policy_paper)
+
+    announcement = build(:statistics_announcement, publication_type_id: PublicationType::OfficialStatistics.id)
+
+    announcement.publication = statistics
+    assert announcement.valid?
+
+    announcement.publication = national_statistics
+    assert_not announcement.valid?
+    assert_equal ["type does not match announcement type: must be 'Official Statistics'"], announcement.errors[:publication]
+
+    announcement.publication_type_id = PublicationType::NationalStatistics.id
+    assert announcement.valid?
+
+    announcement.publication = policy_paper
+    assert_not announcement.valid?
+    assert_equal ["type does not match announcement type: must be 'Accredited Official Statistics'"], announcement.errors[:publication]
+  end
+
+  test "deleting statistics announcement does not delete publication" do
+    national_statistics = create(:draft_national_statistics, title: "Test")
+    announcement = create(:statistics_announcement, publication_type_id: PublicationType::NationalStatistics.id, publication: national_statistics)
+
+    announcement.delete
+    assert_equal Publication.find(national_statistics.id).title, "Test"
+    assert_equal StatisticsAnnouncement.where(id: announcement.id).count, 0
+  end
+
+  # === BEGIN: Publication Type update callback ===
+  test "should allow updating publication type with no attached publication" do
+    announcement = create(:statistics_announcement, publication_type_id: PublicationType::OfficialStatistics.id)
+
+    announcement.publication_type_id = PublicationType::NationalStatistics.id
+    announcement.save!
+
+    assert announcement.valid?
+    assert announcement.errors.empty?
+  end
+
+  test "should allow updating to same publication type with attached draft publication" do
+    national_statistics = create(:draft_national_statistics)
+    announcement = create(:statistics_announcement, publication_type_id: PublicationType::NationalStatistics.id, publication: national_statistics)
+
+    announcement.assign_attributes(
+      publication_type_id: PublicationType::NationalStatistics.id,
+      title: "New title",
+    )
+    announcement.save!
+
+    assert announcement.valid?
+    assert announcement.errors.empty?
+    assert_equal PublicationType::NationalStatistics.id, national_statistics.reload.publication_type_id
+  end
+
+  test "should allow updating to different publication type to attached publication and updates the draft publication" do
```

Comment:
A bit confusing again, maybe try smth like "should update to a different publication type *from* attached publication, and update the draft publication".